src/channel.cpp

```diff
@@ -578,39 +578,39 @@ void Channel::loadChannel(const json_t* channode) {
 
 string Channel::modeToString() {
     switch (chatMode) {
-        default:
         case CMM_BOTH:
             return "both";
         case CMM_ADS_ONLY:
             return "ads";
         case CMM_CHAT_ONLY:
             return "chat";
+        // For now, explicitly defining the default as both.
+        default:
+            return "both";
     }
-    return "both";
 }
 
 ChannelMessageMode Channel::stringToMode(string modestring) {
-    if (modestring == "both")
-        return CMM_BOTH;
-    else if (modestring == "ads")
+    if (modestring == "ads")
```

Comment:
Wouldn't this be better as a switch too? essentially you have made the same change here as the other two.

Reply:
Unfortunately, switch is incapable of dealing with std::string comparisons, hence why the if/else if occurs on these comparisons. Otherwise I'd happily change this to be switch as well. xD